### PR TITLE
Add encoding/precision hints and filtering by category/field name

### PR DESCRIPTION
**Pull request by JonStargaryen**
Additional changes:

- defines `mmtf-filter.json` to match data provided by MMTF
- temporary switch to local molstar distribution for development

### CODE REVIEW

File: src/apps/cif2bcif/index.ts [hunk 1]
```diff
@@ -28,9 +45,20 @@ parser.addArgument([ 'src' ], {
 parser.addArgument([ 'out' ], {
     help: 'Output BCIF path'
 });
+parser.addArgument([ '-config' ], {
```
Comment (arose) [inline src/apps/cif2bcif/index.ts]: I usually do a short '-c' and a long '--config' argument version

File: package.json [hunk 2]
```diff
@@ -40,14 +40,15 @@
     "@types/node": "^12.0.4",
     "@types/node-fetch": "^2.3.4",
     "concurrently": "^4.1.0",
-    "molstar": "^0.2.2",
+    "molstar": "../molstar",
```
Comment (arose) [inline package.json]: Don't commit the local path. I will accept the molstar PR #17 and do a new release to reference

File: package.json [hunk 3]
```diff
@@ -40,14 +40,15 @@
...
     "node-fetch": "^2.6.0",
     "tslint": "^5.17.0",
     "typescript": "3.5.1",
     "webpack": "^4.32.2",
     "webpack-cli": "^3.3.2"
   },
   "dependencies": {
-    "argparse": "^1.0.10"
+    "argparse": "^1.0.10",
+    "util.promisify": "^1.0.0"
```
Comment (arose) [inline package.json]: please also add this to webpack as an external package `'util.promisify': 'require("util.promisify")',`

File: src/apps/cif2bcif/converter.ts [hunk 4]
```diff
@@ -4,20 +4,45 @@
  * @author David Sehnal <david.sehnal@gmail.com>
```
Comment (arose) [inline src/apps/cif2bcif/converter.ts]: don't forget to add you name to the authors of a file if you contributed substantially

File: data/mmtf-filter.json [hunk 5]
```diff
@@ -0,0 +1,272 @@
+[{
+    "categoryName": "cell",
+    "columnName": "length_a",
+    "behavior": "whitelist"
```
Comment (arose) [inline data/mmtf-filter.json]: Have you thought about using "whitelist" as a default for "behavior"? That could make the config less verbose and I would argue that users either supply a large list of whitelisted or a small list of blacklisted categories.

File: src/apps/cif2bcif/index.ts (hunk 1 above)
Comment (JonStargaryen) [reply src/apps/cif2bcif/index.ts]: Done in 956fc60

**Comment by JonStargaryen** [comment]
> Could you please remove the data/ccb.bcif, we don't need it here and it is very large for the repository

Forgot about that one - removed.

File: data/mmtf-filter.json (hunk 5 above)
Comment (JonStargaryen) [reply data/mmtf-filter.json]: Simplified, retained the functionality though. See #17 in molstar.

File: package.json (hunk 2 above)
Comment (JonStargaryen) [reply package.json]: Reverted to original definition

File: package.json (hunk 3 above)
Comment (JonStargaryen) [reply package.json]: Done in 7573d26

File: src/apps/cif2bcif/converter.ts (hunk 4 above)
Comment (JonStargaryen) [reply src/apps/cif2bcif/converter.ts]: Fair enough. Normally, I don't do that as the git history/blame points to authors reliably. Added in 9b85008

**Comment by arose** [comment]
Hey @JonStargaryen, I just published molstar v0.2.3 so you can update the package.json here.

**Comment by JonStargaryen** [comment]
> Hey @JonStargaryen, I just published molstar v0.2.3 so you can update the package.json here.

Thanks @arose - done in latest commit.

**Pull request merged**